File: java/src/5.java
class problem5 {
    public static void main(String[] args){
        long number = 1L;
        while (!isDivisible(number)) {
            number += 1;
        }
        System.out.println(String.valueOf(number));
    }

    public static boolean isDivisible(long num) {
        for (int i = 1; i <= 20; i++) {
            if (num % i != 0)
                return false;
        }
        return true;
    }
}

File: c++/src/5.cpp
#include <stdio.h>
#include <math.h>
#include <string>
#include <iostream>
#include <algorithm>

using namespace std;

int isDivisible(long long num){
    // checks if no can be divided by every number up to 20
    for (int i = 1; i <= 20; i++) {
        if (num % i != 0)
            return false;
    }
    return true;

}

int main(int argc, char *argv[])
{
    long long number = 1;
    while (!isDivisible(number)) {
        number +=1;
    }
    cout << number << "\n";
    return 0;
}
